### PR TITLE
Add disabled_capabilities settings for hover and completion

### PR DESCRIPTION
**Pull request by tomv564**
For #629 and #574

We should probably move the colorProvider toggle into this as well.

### CODE REVIEW

**Comment by tomv564** [comment]
Added colorProvider and documentHighlight as both have existing ways of being turned off in the settings.
We should remove the show_color_box setting before the next release, perhaps part of fixing #703

File: plugin/color.py [hunk 1]
```diff
@@ -39,9 +39,11 @@ def __init__(self, view: sublime.View) -> None:
 
     @classmethod
     def is_applicable(cls, _settings):
-        if settings.show_color_box:
-            return True
-        return False
+        if 'colorProvider' in settings.disabled_capabilities:
+            return False
+        elif not settings.show_color_box:
```
Comment (rwols) [inline plugin/color.py]: Perhaps just remove this setting?

File: docs/features.md [hunk 2]
```diff
@@ -82,6 +82,7 @@ Global plugin settings and settings defined at project level are merged together
 * `diagnostics_gutter_marker` `"dot"` *gutter marker for code diagnostics: "dot", "circle", "bookmark", "cross" or ""*
 * `show_code_actions_bulb` `false` *show a bulb in the gutter when code actions are available*
 * `show_color_box` `true` *Show a box next to the color if the language server provides color support*
+* `disabled_capabilities`, `[]` *Turn off client capabilities (features), like "hover" or "completion"
```
Comment (rwols) [inline docs/features.md]: Please clarify that valid identifiers are coming from `TextDocumentClientCapabilities`.

Comment (tomv564) [reply docs/features.md]: Good point, added all valid options here. Currently the subset is fairly small, so I don't want to mislead people into thinking all the capabilities can be toggled this way.

File: plugin/color.py (hunk 1 above)
Comment (tomv564) [reply plugin/color.py]: Done!

**Pull request merged**